src/main/java/org/reactome/util/general/MandatoryProperties.java

```diff
@@ -0,0 +1,79 @@
+package org.reactome.util.general;
+
+import java.util.Properties;
+
+/**
+ * A specialised implementation of Properties that will throw an exception if a requested property key is missing or has no value. 
+ * @author sshorser
+ *
+ */
+public class MandatoryProperties extends Properties
+{
+	/**
+	 * This exception indicates that a Property key was not present.
+	 * @author sshorser
+	 *
+	 */
+	public class PropertyNotPresentException extends RuntimeException
```

Comment:
I'm not sure, but would this be better to extend `Exception` so it is a checked exception?

Reply:
If I do that, I can't override `getProperty`, since `getProperty` in `Properties` does not throw anything.

Reply:
Okay

Reply:
...but I was able to make `PropertyHasNoValueException` extend `Exception`.

Reply:
Okay, that's good :).  I'm just wondering, though, if it's better to leave them both as extending RuntimeException if one has to, so they are consistent?

It might be confusing why one is checked and one is unchecked?

Reply:
Ah, ok I'll put it back.